markdown/plext/using_pljava.html.md.erb

```diff
@@ -299,411 +331,770 @@ Scalar types are mapped in a straightforward way. This table lists the current m
-     RETURNS SETOF complexTest
-     AS 'foo.fee.Fum.listComplexTest'
-   IMMUTABLE LANGUAGE java;
-```
+The HAWQ query evaluator calls `assignRowValues()` repeatedly until it returns false or until the evaluator decides that it does not need any more rows. At that point, it will call `close()`.
+
+Example: Using the `ResultSetProvider` Interface
+
+1. Create a new file named `FumSetOfComplex.java`, adding the following text to create a class named `FumSetOfComplex` that implements the `ResultSetProvider` interface:
+
+    ``` shell
+    $ cd $PLJAVAWORK/pljex/foo/fee
+    $ vi FumSetOfComplex.java
+    ```
+
+    ``` java
+    package foo.fee;
+    import java.sql.ResultSet;
+    import java.sql.SQLException;
+    import java.sql.Timestamp;
+    import org.postgresql.pljava.ResultSetProvider;
 
-The function maps to a static java method that returns an instance that implements the `ResultSetProvider` interface.
-
-```java
-public class Fum implements ResultSetProvider
-{
-  private final int m_base;
-  private final int m_increment;
-  public Fum(int base, int increment)
-  {
-    m_base = base;
-    m_increment = increment;
-  }
-  public boolean assignRowValues(ResultSet receiver, int 
+    public class FumSetOfComplex implements ResultSetProvider
+    {
+      private final int m_base;
+      private final int m_increment;
+      public FumSetOfComplex(int base, int increment)
+      {
+        m_base = base;
+        m_increment = increment;
+      }
+      public boolean assignRowValues(ResultSet receiver, int 
 currentRow)
-  throws SQLException
-  {
-    // Stop when we reach 12 rows.
-    //
-    if(currentRow >= 12)
-      return false;
-    receiver.updateInt(1, m_base);
-    receiver.updateInt(2, m_base + m_increment * currentRow);
-    receiver.updateTimestamp(3, new 
+      throws SQLException
+      {
+        if(currentRow >= 12)
+          return false;
+        receiver.updateInt(1, m_base);
+        receiver.updateInt(2, m_base + m_increment * currentRow);
+        receiver.updateTimestamp(3, new 
 Timestamp(System.currentTimeMillis()));
-    return true;
-  }
-  public void close()
-  {
-   // Nothing needed in this example
-  }
-  public static ResultSetProvider listComplexTests(int base, 
+        return true;
+      }
+      public void close()
+      {
+         /* Nothing needed in this example */
+      }
+      public static ResultSetProvider listComplex(int base, 
 int increment)
-  throws SQLException
-  {
-    return new Fum(base, increment);
-  }
-}
-```
+      throws SQLException
+      {
+        return new FumSetOfComplex(base, increment);
+      }
+    }
+    ```
+
+    The `listComplex()` method is called once. It will return NULL if no results are available or an instance of the `ResultSetProvider` interface. The Java class `FumSetOfComplex` implements this interface to return an instance of itself. `assignRowValues()` is called repeatedly until it returns false. At that time, `close()` is called.
```

Comment:
This sentence isn't complete.  "... or an instance of the `ResultSetProvider` interface <something>."